src/main/java/com/wolox/training/models/User.java

```diff
@@ -0,0 +1,94 @@
+package com.wolox.training.models;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.wolox.training.exceptions.BookAlreadyOwnedException;
+import com.wolox.training.exceptions.NotFoundException;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Entity(name = "Users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @NotNull
+    private String username;
+
+    @NotNull
+    private String name;
+
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern="yyyy-MM-dd")
+    @NotNull
+    private LocalDate birthdate;
+
+    @ManyToMany(cascade = CascadeType.ALL)
```

Comment:
The CascadeType should be MERGE and REFRESH. Here's why:
You have [different ways of cascading](https://www.baeldung.com/jpa-cascade-types). By saying ALL, you are saying: Persist, Merge, Refresh, Delete and Detach.

Detach removes the entity from the persistent context. When we use CascaseType.DETACH, the child entity will also get removed from the persistent context. (yes I copied the baeldung explanation because detached it's almost never used and I don't know how to explain it better 😋)

- Persist says: If you persist the Users instance to the DB, then persist every item that this collection has in the DB as well. In this case, you would be also persisting all the books contained by a user, we don't need that, we just want to assign books that already exist in de DB.
- Delete says: If you delete the Users instance from the DB, then delete every item that this collection has in the DB as well. In this case, you would be deleting all the user's books, we don't want that, because a book could be contained by other users who still need that book.
- Merge says: If you changed the in-memory values of the collection, then merge this changes to the DB so these are reflected there. We want this in order to have the same information in-memory and in the bd, it provides us more consistency in our information.
- Refresh says: If something changed in the database values of the collection, then refresh those changes in the in-memory values. It is the inverse way of the _merge_ cascade type. It also provides consistency, so we want this to happen.